### PR TITLE
PYIC-6243: Add CheckMobileAppVcReceiptHandler

### PR DESCRIPTION
**Pull request by MikeCollingwood**
## Proposed changes

- Add CheckMobileAppVcReceiptHandler lambda

## Why did it change

- To allow the frontend to poll for mobile app receipt

### Issue tracking

- [PYIC-6243](https://govukverify.atlassian.net/browse/PYIC-6243)

See [POC](https://github.com/govuk-one-login/ipv-core-back/pull/2644) for removing the need for [PYIC-6244](https://govukverify.atlassian.net/browse/PYIC-6244).

[PYIC-6243]: https://govukverify.atlassian.net/browse/PYIC-6243?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PYIC-6244]: https://govukverify.atlassian.net/browse/PYIC-6244?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: openAPI/core-back-internal.yaml [hunk 1]
```diff
@@ -60,6 +60,24 @@ paths:
           Fn::Sub: arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${ProcessMobileAppCallbackFunction.Arn}:live/invocations
         passthroughBehavior: "when_no_match"
 
+  /app/check-vc-receipt:
+    get:
+      description: |
+        Called each poll of the backend by the frontend when checking receipt of the verifiable credentials associated to a mobile app journey. This triggers CRI response and verifiable credential retrieval.
+      responses:
+        100:
```
Comment (MikeCollingwood) [inline openAPI/core-back-internal.yaml:68]: Need to update to 404

File: libs/common-services/src/test/java/uk/gov/di/ipv/core/library/fixtures/TestFixtures.java [hunk 2]
```diff
@@ -185,23 +181,4 @@ static JWEObject createJweObject(RSAEncrypter rsaEncrypter, SignedJWT signedJWT)
             throw new HttpResponseExceptionWithErrorBody(500, ErrorResponse.FAILED_TO_ENCRYPT_JWT);
         }
     }
-
```
Comment (MikeCollingwood) [inline libs/common-services/src/test/java/uk/gov/di/ipv/core/library/fixtures/TestFixtures.java:188]: Unused

File: settings.gradle [hunk 3]
```diff
@@ -61,3 +61,5 @@ dependencyResolutionManagement {
 		}
 	}
 }
+include 'lambdas:check-mobile-app-vc-receipt'
+findProject(':lambdas:check-mobile-app-vc-receipt')?.name = 'check-mobile-app-vc-receipt'
```
Comment (MikeCollingwood) [inline settings.gradle:65]: Fix this

File: lambdas/check-mobile-app-vc-receipt/src/main/java/uk/gov/di/ipv/core/checkmobileappvcreceipt/CheckMobileAppVcReceiptHandler.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        this.verifiableCredentialService = verifiableCredentialService;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public CheckMobileAppVcReceiptHandler() {
+        configService = ConfigService.create();
+        ipvSessionService = new IpvSessionService(configService);
+        clientOAuthSessionDetailsService = new ClientOAuthSessionDetailsService(configService);
+        criResponseService = new CriResponseService(configService);
+        this.verifiableCredentialService = new VerifiableCredentialService(configService);
```
Comment (MikeCollingwood) [inline lambdas/check-mobile-app-vc-receipt/src/main/java/uk/gov/di/ipv/core/checkmobileappvcreceipt/CheckMobileAppVcReceiptHandler.java:61]:
```suggestion
        verifiableCredentialService = new VerifiableCredentialService(configService);
```

File: lambdas/check-mobile-app-vc-receipt/src/main/java/uk/gov/di/ipv/core/checkmobileappvcreceipt/CheckMobileAppVcReceiptHandler.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        verifiableCredentialService = new VerifiableCredentialService(configService);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input, Context context) {
+        try {
+            var callbackRequest = parseCallbackRequest(input);
```
Comment (MikeCollingwood) [inline lambdas/check-mobile-app-vc-receipt/src/main/java/uk/gov/di/ipv/core/checkmobileappvcreceipt/CheckMobileAppVcReceiptHandler.java:70]: rename

File: lambdas/check-mobile-app-vc-receipt/src/test/java/uk/gov/di/ipv/core/processmobileappcallback/CheckMobileAppVcReceiptHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,179 @@
+package uk.gov.di.ipv.core.processmobileappcallback;
```
Comment (MikeCollingwood) [inline lambdas/check-mobile-app-vc-receipt/src/test/java/uk/gov/di/ipv/core/processmobileappcallback/CheckMobileAppVcReceiptHandlerTest.java:1]: fix path

**Review by blakeyp — APPROVED** [review]
🎉

**Pull request merged**